convert2rhel/backup/files.py

```diff
@@ -0,0 +1,153 @@
+            return
+
+        loggerinst.info("Backing up %s." % self.filepath)
+        if os.path.isfile(self.filepath):
+            try:
+                shutil.copy2(self.filepath, BACKUP_DIR)
+                loggerinst.debug("Copied %s to %s." % (self.filepath, BACKUP_DIR))
+            except (OSError, IOError) as err:
+                # IOError for py2 and OSError for py3
+                loggerinst.critical_no_exit("Error(%s): %s" % (err.errno, err.strerror))
+                raise exceptions.CriticalError(
+                    id_="FAILED_TO_SAVE_FILE_TO_BACKUP_DIR",
+                    title="Failed to copy file to the backup directory.",
+                    description=(
+                        "Copying the current file has failed. This can lead to inconsistency during the rollbacks as "
+                        "convert2rhel won't be able to restore the file in case of failures."
+                        "In the current case, we encountered a failure while performing that backup so it is unsafe "
+                        "to continue. See the diagnosis section to identify which problem ocurred during the backup."
+                    ),
+                    diagnosis="Failed to backup %s. Errno: %s, Error: %s" % (self.filepath, err.errno, err.strerror),
+                )
+        else:
+            loggerinst.info("Can't find %s.", self.filepath)
+
+        # Set the enabled value
+        super(RestorableFile, self).enable()
+
+    def restore(self, rollback=True):
+        """Restore a previously backed up file"""
+        if rollback:
+            loggerinst.task("Rollback: Restore %s from backup" % self.filepath)
+        else:
+            loggerinst.info("Restoring %s from backup" % self.filepath)
+
+        backup_filepath = os.path.join(BACKUP_DIR, os.path.basename(self.filepath))
+
+        # We do not have backup or not backed up by this
+        if not self.enabled or not os.path.isfile(backup_filepath):
+            loggerinst.info("%s hasn't been backed up." % self.filepath)
+            return
+
+        try:
+            shutil.copy2(backup_filepath, self.filepath)
+        except (OSError, IOError) as err:
+            # Do not call 'critical' which would halt the program. We are in
+            # a rollback phase now and we want to rollback as much as possible.
+            # IOError for py2 and OSError for py3
+            loggerinst.critical_no_exit("Error(%s): %s" % (err.errno, err.strerror))
+            return
+
+        if rollback:
+            loggerinst.info("File %s restored." % self.filepath)
+            super(RestorableFile, self).restore()
+        else:
+            loggerinst.debug("File %s restored." % self.filepath)
+            # not setting enabled to false since this is not being rollback
+            # restoring the backed up file for conversion purposes
+
+    # Probably will be deprecated and unusable since using the BackupController
+    # Depends on specific usage of this
```

Comment:
```suggestion

```
I think this comment can be removed since it's still used in the right way in `register_system()`. The fix was to [RHELC-16](https://issues.redhat.com/browse/RHELC-16).